GitUtils.ps1

```diff
@@ -21,23 +21,20 @@ function Get-GitDirectory {
         $null
     }
     elseif ($Env:GIT_DIR) {
-        $Env:GIT_DIR
+        $Env:GIT_DIR -replace '\\|/', [System.IO.Path]::DirectorySeparatorChar
     }
     else {
         if (Test-Path .git) {
             [System.IO.Path]::Combine($pathInfo.Path, '.git')
         }
         else {
-            # A native executable that writes to stderr AND redirects stderr will generate non-terminating
-            # error records if $ErrorActionPreference is set to Stop.
-            $ErrorActionPreference = 'Continue'
-            $gitDir = git rev-parse --git-dir 2>$null
+            $gitDir = Invoke-Utf8ConsoleCommand { git rev-parse --git-dir 2>$null }
             if ($gitDir -eq '.') {
                 # If in root or bare repo, we get '.' back
                 $pathInfo.Path
             }
             else {
-                $gitDir
+                $gitDir -replace '\\|/', [System.IO.Path]::DirectorySeparatorChar

```

Comment:
Would it make sense to use `Resolve-Path` here instead? That would also eliminate the need to handle `.` separately.